src/Groups/group_constructors.jl

```diff
@@ -0,0 +1,446 @@
+"""
+function orthogonal_group(n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.GO(n, q))
+end
+
+function orthogonal_group(::Type{T}, n::Int, q::Int) where T <: GAPGroup
+  return T(GAP.Globals.GO(_gap_filter(T), n, q))
+end
+
+function orthogonal_group(e::Int, n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.GO(e, n, q))
+end
+
+function orthogonal_group(::Type{T}, e::Int, n::Int, q::Int) where T <: GAPGroup
+  return T(GAP.Globals.GO(_gap_filter(T), e, n, q))
+end
+
+"""
+    special_orthogonal_group(n::Int, q::Int)
+    special_orthogonal_group(T::Type, n::Int, q::Int)
+    special_orthogonal_group(e::Int, n::Int, q::Int)
+    special_orthogonal_group(::Type{T}, e::Int, n::Int, q::Int)
+    SO = special_orthogonal_group
+return the special orthogonal group of dimension `n` over the field GF(`q`) of type `e`, where `e` in {`+1`,`-1`} for `n` even and `e`=`0` for `n` odd. If `n` is odd, `e` can be omitted. It is returned of type `T` for `T` = `MatrixGroup` or `PermGroup`.
+"""
+function special_orthogonal_group(n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.SO(n, q))
+end
+
+function special_orthogonal_group(::Type{T}, n::Int, q::Int) where T <: GAPGroup
+  return T(GAP.Globals.SO(_gap_filter(T), n, q))
+end
+
+function special_orthogonal_group(e::Int, n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.SO(e, n, q))
+end
+
+function special_orthogonal_group(::Type{T}, e::Int, n::Int, q::Int) where T <: GAPGroup
+  return T(GAP.Globals.SO(_gap_filter(T), e, n, q))
+end
+
+"""
+    omega_group(n::Int, q::Int)
+    omega_group(T::Type, n::Int, q::Int)
+    omega_group(e::Int, n::Int, q::Int)
+    omega_group(::Type{T}, e::Int, n::Int, q::Int)
+return the Omega group of dimension `n` over the field GF(`q`) of type `e`, where `e` in {`+1`,`-1`} for `n` even and `e`=`0` for `n` odd. If `n` is odd, `e` can be omitted. It is returned of type `T` for `T` = `MatrixGroup` or `PermGroup`.
+"""
+function omega_group(n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.Omega(n, q))
+end
+
+function omega_group(::Type{T}, n::Int, q::Int) where T <: GAPGroup
+  return T(GAP.Globals.Omega(_gap_filter(T), n, q))
+end
+
+function omega_group(e::Int, n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.Omega(e, n, q))
+end
+
+function omega_group(::Type{T}, e::Int, n::Int, q::Int) where T <: GAPGroup
+  return T(GAP.Globals.Omega(_gap_filter(T), e, n, q))
+end
+
+"""
+    general_linear_group(n::Int, q::Int)
+    general_linear_group(T::Type, n::Int, q::Int)
+    GL(n::Int, q::Int)
+return the general linear group of dimension `n` over the field GF(`q`). It is returned of type `T` for `T` = `MatrixGroup` or `PermGroup`.
+"""
+GL = general_linear_group
+
+"""
+    special_linear_group(n::Int, q::Int)
+    special_linear_group(T::Type, n::Int, q::Int)
+    SL(n::Int, q::Int)
+return the special linear group of dimension `n` over the field GF(`q`). It is returned of type `T` for `T` = `MatrixGroup` or `PermGroup`.
+"""
+SL = special_linear_group
```

Comment:
```suggestion
const SL = special_linear_group
```
Also, why is this here, and not where `special_linear_group` is defined? And why the duplicate documentation?

The same of course holds for GL, Sp, etc.